module/mfrc522_spi.h

```diff
@@ -19,8 +19,37 @@ struct address_byte {
 #define MFRC522_SPI_WRITE 0
 #define MFRC522_SPI_READ 1
 
+// MFRC522 commands, see 10.3
+#define MFRC522_COMMAND_IDLE 0b0000
+#define MFRC522_COMMAND_MEM 0b0001
+#define MFRC522_COMMAND_GENERATE_RANDOM_ID 0b0010
+#define MFRC522_COMMAND_CALC_CRC 0b0010
+#define MFRC522_COMMAND_TRANSMIT 0b0100
+#define MFRC522_COMMAND_NO_CMD_CHANGE 0b0111
+#define MFRC522_COMMAND_RECEIVE 0b1000
+#define MFRC522_COMMAND_TRANSCEVIE 0b1100
+#define MFRC522_COMMAND_MF_AUTHENT 0b1110
+#define MFRC522_COMMAND_SOFT_RESET 0b1111
+
+// Values for CommandReg, see 9.3.1.2
+#define MFRC522_COMMAND_REG 0x1
+#define MFRC522_COMMAND_REG_RCV_ON 0
+#define MFRC522_COMMAND_REG_RCV_OFF 1
+#define MFRC522_COMMAND_REG_POWER_DOWN_ON 1
+#define MFRC522_COMMAND_REG_POWER_DOWN_OFF 0
+#define MFRC522_COMMAND_REG_RCV_OFF_BIT 5
+#define MFRC522_COMMAND_REG_POWER_DOWN_BIT 4
+#define MFRC522_COMMAND_REG_COMMAND_MASK 0xF
```

Comment:
Is it necessary to expose these defines in the header? I think it would make more sense to simply keep them in the source file instead, maybe even for each function since they're used only once.
Something like
```c
void fifo_flush(void) {
#define ..._FIFO_FLUSH_SHIFT 7

/* Rest of code */

#undef ..._FIFO_FLUSH_SHIFT
}
```

Reply:
In c file ok but I don't know about define in a function directly

Reply:
@Skallwar since we only use them in one place it's equivalent I believe, but as you wish. This way it doesn't pollute the top of the source file but it does pollute function defintions a bit more